addOns/sstiscanner/src/main/java/org/zaproxy/zap/extension/sstiscanner/SSTIScanner.java

```diff
@@ -0,0 +1,501 @@
+
+    /**
+     * Get the unique identifier of this plugin
+     *
+     * @return this plugin identifier
+     */
+    @Override
+    public int getId() {
+        return 90035;
+    }
+
+    /**
+     * Get the name of this plugin
+     *
+     * @return the plugin name
+     */
+    @Override
+    public String getName() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "name");
+    }
+
+    /**
+     * Give back specific plugin dependencies (none for this)
+     *
+     * @return the list of plugins that need to be executed before
+     */
+    @Override
+    public String[] getDependency() {
+        return new String[] {};
+        // TODO activate when start to work properly
+        // return new String[] {"TestPersistentXSSSpider"};
+    }
+
+    @Override
+    public boolean targets(TechSet technologies) {
+        return true;
+    }
+
+    /**
+     * Get the description of the vulnerability when found
+     *
+     * @return the vulnerability description
+     */
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+    }
+
+    /**
+     * Give back the categorization of the vulnerability checked by this plugin (it's an injection
+     * category for CODEi)
+     *
+     * @return a category from the Category enum list
+     */
+    @Override
+    public int getCategory() {
+        return Category.INJECTION;
+    }
+
+    /**
+     * Give back a general solution for the found vulnerability
+     *
+     * @return the solution that can be put in place
+     */
+    @Override
+    public String getSolution() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+    }
+
+    /**
+     * Reports all links and documentation which refers to this vulnerability
+     *
+     * @return a string based list of references
+     */
+    @Override
+    public String getReference() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+    }
+
+    /**
+     * http://cwe.mitre.org/data/definitions/94.html
+     *
+     * @return the official CWE id
+     */
+    @Override
+    public int getCweId() {
+        return 94; // WE-94: Improper Control of Generation of Code ('Code Injection').
+    }
+
+    /**
+     * Seems no WASC defined for this
+     *
+     * @return the official WASC id
+     */
+    @Override
+    public int getWascId() {
+        return 20; // WASC-20: Improper Input Handling
+    }
+
+    /**
+     * Give back the risk associated to this vulnerability (high)
+     *
+     * @return the risk according to the Alert enum
+     */
+    @Override
+    public int getRisk() {
+        return Alert.RISK_HIGH;
+    }
+
+    /** Initialize the plugin according to the overall environment configuration */
+    @Override
+    public void init() {
+        // Tech python = new Tech(Tech.Lang, "Python");
+        // Tech.builtInTech = ArrayUtils.add( Tech.builtInTech, python );
```

Comment:
Is this needed? 
Add a comment explaining?

Reply:
Not anymore.